release-tools/build.make

```diff
@@ -30,7 +30,7 @@ CMDS_DIR ?= cmd
 REGISTRY_NAME?=quay.io/k8scsi
 
 # Can be set to -mod=vendor to ensure that the "vendor" directory is used.
-GOFLAGS_VENDOR=
+GOFLAGS_VENDOR=-mod=vendor
```

Comment:
I think we'd better avoid carry patches in `release-tools/`. It's a subtree of https://github.com/kubernetes-csi/csi-release-tools and adding direct commits causes `./release-tools/verify-subtree.sh release-tools/` to fail. In some projects, verify-subtree.sh runs as part of `make verify` but in this case... it fails even without your change :)

`make smb` already includes [-mod vendor](https://github.com/openshift/csi-driver-smb/blob/master/Makefile#L125)... could you use that Makefile target in the Dockerfile and avoid the carry patch?

Reply:
The [azure-disk-csi-driver Dockerfile](https://github.com/openshift/azure-disk-csi-driver/blob/e5c4b2a256181e7265d8cf2ccc5760289f9d48bd/Dockerfile.openshift.rhel7#L4) has a good example, the `ARCH=$(go env GOARCH)` was needed to avoid [always trying to build amd64](https://github.com/openshift/azure-disk-csi-driver/blob/e5c4b2a256181e7265d8cf2ccc5760289f9d48bd/Makefile#L80).